Auth.php

```diff
@@ -495,7 +495,10 @@ protected function addSession(int $uid, bool $remember)
             'samesite' => $this->config->cookie_samesite ?? 'Lax' // None || Lax  || Strict
         ];
 
-        setcookie($this->config->cookie_name, $data['hash'], $cookie_options);
+        if(!setcookie($this->config->cookie_name, $data['hash'], $cookie_options)){
+            return false;
+        }
+
 
         $_COOKIE[$this->config->cookie_name] = $data['hash'];
```

Comment:
is this line necessary? i know it was there before; is it probably for php, validating the session again?

Reply:
which line?

Reply:
Line 500

Reply:
This may be a vulnerability, because the cookie probably was not set - because of security reasons of the browser ?!

Reply:
oh, was looking at the wrong line.. That line doesn't make sense no, that line need to be removed..

Reply:
> 
> 
> This may be a vulnerability, because the cookie probably was not set - because of security reasons of the browser ?!

Not sure what you mean?

Reply:
You could see the setting of the cookie as necessary part of an successful login - if the cookie could not be set, the login is invalid.

Reply:
correct, needs to return invalid instead of false

Reply:
How would you return invalid? Like throw an exception?

Reply:
Like it is here: https://github.com/PHPAuth/PHPAuth/blob/master/Auth.php#L161